diff --git a/ci/install-nnpy.sh b/ci/install-nnpy.sh
index 558d2a0..b7d0d76 100644
--- a/ci/install-nnpy.sh
+++ b/ci/install-nnpy.sh
@@ -1,3 +1,3 @@
 #!/bin/sh
 set -e
-sudo pip3 install nnpy cffi
+pip3 install nnpy cffi
diff --git a/ci/install-thrift.sh b/ci/install-thrift.sh
index c3f7a2e..1805da3 100644
--- a/ci/install-thrift.sh
+++ b/ci/install-thrift.sh
@@ -3,17 +3,13 @@
 THIS_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 source $THIS_DIR/common.sh
 
-check_lib libthrift libthrift-0.13.0
+check_lib libthrift libthrift-0.16.0
 
 set -e
-# Make it possible to get thrift in China
-# wget http://archive.apache.org/dist/thrift/0.13.0/thrift-0.13.0.tar.gz
-# tar -xzvf thrift-0.13.0.tar.gz
-git clone -b 0.13.0 https://github.com/apache/thrift.git thrift-0.13.0
-cd thrift-0.13.0
+git clone -b 0.16.0 https://github.com/apache/thrift.git thrift-0.16.0
+cd thrift-0.16.0
 ./bootstrap.sh
-./configure --with-cpp=yes --with-c_glib=no --with-java=no --with-ruby=no --with-erlang=no --with-go=no --with-nodejs=no
+./configure --with-cpp=yes --with-c_glib=no --with-java=no --with-ruby=no --with-erlang=no --with-go=no --with-nodejs=no --with-python=no
 make -j2 && sudo make install
-cd lib/py
-sudo python3 setup.py install
-cd ../../..
+cd ..
+pip install thrift==0.16.0
