diff --git a/util/install.sh b/util/install.sh
index 0a67871..90070a5 100755
--- a/util/install.sh
+++ b/util/install.sh
@@ -5,6 +5,7 @@
 
 # Fail on error
 set -e
+set -x
 
 # Fail on unset var usage
 set -o nounset
@@ -180,9 +181,10 @@ function mn_deps {
                 pf=pyflakes3
         fi
         $install gcc make socat psmisc xterm ssh iperf telnet \
-                 ethtool help2man $pf pylint pep8 \
+                 ethtool help2man pylint3 pep8 \
                  net-tools \
                  ${PYPKG}-pexpect ${PYPKG}-tk
+        $install --no-install-recommends $pf
         # Install pip
         $install ${PYPKG}-pip || $install ${PYPKG}-pip-whl
         if ! ${PYTHON} -m pip -V; then
