diff --git a/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp b/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
index 0b2322722..8d0d33834 100644
--- a/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
+++ b/backends/p4tools/modules/testgen/targets/bmv2/expr_stepper.cpp
@@ -880,7 +880,7 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      testBackend);
                  auto &nextState = state.clone();
                  nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            BMv2Constants::METER_COLOR::GREEN));
+                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;
@@ -1002,7 +1002,7 @@ void Bmv2V1ModelExprStepper::evalExternMethodCall(const IR::MethodCallExpression
                      "(GREEN).",
                      testBackend);
                  nextState.set(meterResult, IR::getConstant(meterResult->type,
-                                                            BMv2Constants::METER_COLOR::GREEN));
+                                                            static_cast<big_int>(BMv2Constants::METER_COLOR::GREEN)));
                  nextState.popBody();
                  result->emplace_back(nextState);
                  return;
