diff --git a/utils/Makefile b/utils/Makefile
index 0f32642..79af46f 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -32,10 +32,10 @@ endif
 all: run
 
 run: build
-	sudo python3 $(RUN_SCRIPT) -t $(TOPO) $(run_args)
+	sudo PATH=$(PATH) PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python python3 $(RUN_SCRIPT) -t $(TOPO) $(run_args)
 
 stop:
-	sudo mn -c
+	sudo PATH=$(PATH) `which mn` -c
 
 build: dirs $(compiled_json)
 
